examples/rewrites/app/about/page.tsx

```diff
@@ -0,0 +1,29 @@
+"use client";
+import { useState, useEffect } from "react";
+import { usePathname } from "next/navigation";
+import Link from "next/link";
+import styles from "../../styles.module.css";
+import Code from "../_components/Code";
+
+export default function About() {
+  const pathname = usePathname();
+  const [path, setPath] = useState<string | null>(null);
+
+  useEffect(() => {
+    setPath(pathname);
```

Comment:
It's useless and you can use `pathname` directly in the markup.

Reply:
@siarheipashkevich Feel free to just push up a <picture data-single-emoji=":pr:" title=":pr:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/pr/876af7e7d65e1595.png" alt=":pr:" align="absmiddle"></picture> next time!

- https://github.com/vercel/next.js/pull/69631

Reply:
@siarheipashkevich Actually, this step is necessary to avoid hydration issues when used with rewrites → https://github.com/vercel/next.js/pull/69631#pullrequestreview-2278118356.